core/src/main/kotlin/endpoints/AgentManager.kt

```diff
@@ -108,6 +108,22 @@ class AgentManager(override val kodein: Kodein) : KodeinAware {
         } else null
     }
 
+    //TODO EPMDJ-9872 Add ability to remove offline agent
+    internal suspend fun removeAgent(
+        agentId: String,
+    ): Boolean = (get(agentId) ?: loadAgentInfo(agentId)).let { storedInfo ->
+        val agentStatus = storedInfo?.agentStatus
+        if (storedInfo == null || agentStatus == AgentStatus.NOT_REGISTERED || agentStatus == AgentStatus.PREREGISTERED) {
```

Comment:
storedInfo  == null
what is the case?

Reply:
This is for the case when we don't have AgentInfo, but there is a PreparedAgentData.